Namespace deletion

jerabekjiri opened the pull request:
Issue: [AAH-714](https://issues.redhat.com/browse/AAH-714)

- Added`DeleteCollectionModal` that could be used for deleting collections, namespaces and containers
- With working backend for deleting namespace, I added logic for deleting namespace on UI
- Namespace can be deleted only if no collections in the namespace

- Delete namespace modal
![Screenshot from 2021-09-03 00-36-08](https://user-images.githubusercontent.com/19647757/131925837-7778200b-07ef-459f-b229-64d5b9a9119b.png)

- Error alert if something went wrong
![Screenshot from 2021-09-07 14-31-35](https://user-images.githubusercontent.com/19647757/132346732-8d298ef5-5da4-469d-906d-eba473cc4bbc.png)

- Success alert 
![Screenshot from 2021-09-07 14-24-02](https://user-images.githubusercontent.com/19647757/132346746-f3581856-21f8-4ecf-a9a7-4612f736867b.png)
 
 Success alert is shown on the namespace list page, the flow goes like that: 
1. DELETE namespace 
2. If success -> redirect to namespacelist
3. show success alert

ZitaNemeckova commented:
I have a `arista` namespace with a published collection. When a go to arita detail page and change a repository from `pusblished` to `community` I can try to delete it but it fails with `{"errors":[{"status":"400","code":"invalid","title":"Invalid input.","detail":"Namespace arista cannot be deleted because there are still collections associated with it."}]}`

ZitaNemeckova commented on src/containers/collection-detail/delete-collection-modal.tsx:
```diff
@@ -0,0 +1,123 @@
+
+  private deleteCollection(collection) {}
+
+  private deleteNamespace({ name }) {
```
It would be better to move this function to the container and pass it as a prop.

ZitaNemeckova commented on src/containers/collection-detail/delete-collection-modal.tsx:
```diff
@@ -0,0 +1,123 @@
+            : this.deleteCollection(collection)
+        }
+        title={
+          namespace
```
It would be better to pass title and description in params so the component as dumb as possible.

ZitaNemeckova reviewed (changes requested):
It's on a good way :) There's a fwe things that need rework.
- There' s "collection deletion" logic that should come in it's own PR. 
- Any component should be as dumb as possible. The logic should stay in a container if possible. 
- Why not use `DeleteModal` component instead of create a new component?

jerabekjiri commented:
- Created a new component `DeleteModalWithConfirm` which uses `DeleleModal`, but already includes confirmation UI part and logic for that, this could be nice for reusing later, since all content deletion modals are same and could be reused like that :)
- Fixed mentioned error by getting all namespace's repositories and checking if it has collections.
- Removed collection deletion leftovers
> Any component should be as dumb as possible. The logic should stay in a container if possible.

Good suggestion, thanks, It really made no sense mixing collection and namespace into one delete modal, also when there's container deletion and might others in future :grin: :+1:

jerabekjiri commented:
I will investigate that test fail more, my Chrome crashes on this local test, but in Firefox it passes normally.

himdel commented on src/api/namespace.ts:
```diff
@@ -6,6 +6,10 @@ class API extends HubAPI {
     super();
   }
+
+  deleteNamespace(name) {
```
`NamespaceAPI.delete(name)` already exists from the base class

ZitaNemeckova reviewed (approved):
LGTM 👍 There's a style that should be moved to sccs file in followup PR.
